test/assignment.coffee

```diff
@@ -278,18 +278,17 @@ test "#1024: destructure empty assignments to produce javascript-like results",
 
 test "#1005: invalid identifiers allowed on LHS of destructuring assignment", ->
   disallowed = ['eval', 'arguments'].concat CoffeeScript.RESERVED
-  throws (-> CoffeeScript.compile "[#{disallowed.join ', '}] = x"), null, 'all disallowed'
-  throws (-> CoffeeScript.compile "[#{disallowed.join '..., '}...] = x"), null, 'all disallowed as splats'
+  throwsCompileError "[#{disallowed.join ', '}] = x", null, null, 'all disallowed'
```

Comment:
The pattern is basically that I replaced `throws -> CoffeeScript.compile ...` with `throwsCompileError ...`, which checks that the error gets thrown both during compile-to-JS and during AST generation

And similarly with `doesNotThrow -> CoffeeScript.compile ...` -> `doesNotThrowCompileError ...`